src/VisualStudio/Xaml/Impl/Implementation/XamlProjectService.cs

```diff
@@ -116,21 +140,21 @@ private void EnsureDocument(string filePath)
             {
                 project.AddSourceFile(filePath);
 
-                var documentId = _workspace.CurrentSolution.GetDocumentIdsWithFilePath(filePath).First(d => d.ProjectId == project.Id);
+                var documentId = _workspace.CurrentSolution.GetDocumentIdsWithFilePath(filePath).Single(d => d.ProjectId == project.Id);
                 var document = _workspace.CurrentSolution.GetDocument(documentId)!;
                 var hasText = document.TryGetText(out var text);
                 if (!hasText || text?.Container.TryGetTextBuffer() == null)
```

Comment:
Can you add a link to https://github.com/dotnet/roslyn/issues/49879 to ensure we clean this up since this is working around that problem?